relayer/rpc_relayer.go

```diff
@@ -42,63 +41,33 @@ func (r *RpcRelayer) GetProvider() *ethrpc.Provider {
 	return r.provider
 }
 
-// ..
 func (r *RpcRelayer) EstimateGasLimits(ctx context.Context, walletConfig sequence.WalletConfig, walletContext sequence.WalletContext, txns sequence.Transactions) (sequence.Transactions, error) {
-	// NOTE: this is a stub
-	// TODO: replace this impl with the new impl or call to the server, etc.
-
 	walletAddress, err := sequence.AddressFromWalletConfig(walletConfig, walletContext)
 	if err != nil {
 		return nil, err
 	}
 
-	provider := r.GetProvider()
+	data, err := txns.EncodeRaw()
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := r.protoConfig(ctx, &walletConfig, walletAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := r.Service.UpdateMetaTxnGasLimits(ctx, walletAddress.Hex(), config, hexutil.Encode(data))
+	if err != nil {
+		return nil, err
+	}
 
-	isWalletDeployed, err := sequence.IsWalletDeployed(provider, walletAddress)
+	data, err = hexutil.Decode(response)
```

Comment:
Small pet peeve, but maybe not override data and use a different variable name instead